rust/worker/src/execution/operators/get_segment_writer.rs

```diff
@@ -0,0 +1,154 @@
+use chroma_blockstore::provider::BlockfileProvider;
+use chroma_error::ChromaError;
+use chroma_index::hnsw_provider::HnswIndexProvider;
+use chroma_segment::{
+    blockfile_metadata::{MetadataSegmentError, MetadataSegmentWriter},
+    blockfile_record::{RecordSegmentWriter, RecordSegmentWriterCreationError},
+    distributed_hnsw::{DistributedHNSWSegmentFromSegmentError, DistributedHNSWSegmentWriter},
+    types::ChromaSegmentWriter,
+};
+use chroma_sysdb::{GetCollectionsError, SysDb};
+use chroma_system::Operator;
+use chroma_types::{Segment, SegmentType};
+use thiserror::Error;
+use tonic::async_trait;
+
+#[derive(Debug, Default)]
+pub struct GetSegmentWriterOperator {}
+
+impl GetSegmentWriterOperator {
+    pub fn new() -> Self {
+        GetSegmentWriterOperator::default()
+    }
+}
+
+#[derive(Debug)]
+pub struct GetSegmentWriterInput {
+    blockfile_provider: BlockfileProvider,
+    hnsw_provider: HnswIndexProvider,
+    sysdb: SysDb,
+    segment: Segment,
+}
+
+impl GetSegmentWriterInput {
+    pub fn new(
+        blockfile_provider: BlockfileProvider,
+        hnsw_provider: HnswIndexProvider,
+        sysdb: SysDb,
+        segment: Segment,
+    ) -> Self {
+        GetSegmentWriterInput {
+            blockfile_provider,
+            hnsw_provider,
+            sysdb,
+            segment,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct GetSegmentWriterOutput {
+    pub writer: ChromaSegmentWriter<'static>,
```

Comment:
Enforcing static here is somewhat unfortunate

Reply:
yeah, the lifetime is only used for the metadata segment writer which is already 'static in the current impl on main